### PR TITLE
🐝 Allow unbalanced HTML in markdown card.

### PR DESCRIPTION
**Pull request by disordinary**
No issue.

The simpledom interpreter that the Mobiledoc DOM renderer uses does not allow for unbalanced or incorrect HTML such as that which is entered by a user.

This PR adds a step where the HTML is sanitised and balanced before being passed to simpledom.

### CODE REVIEW

**Comment by kirrg001** [comment]
PR will need to stay open for the moment - it or something similar will be needed to prevent the server-side exploding if there's any invalid HTML added in a markdown or embed card - cc @kevinansfield

**Comment by kevinansfield** [comment]
@kirrg001 I've rebased this on top of the `markdown-it` PR and updated/added comments. It will need merging after the markdown-it PR is merged so that the editor can be used when writing raw HTML without it throwing errors.

It seems like overkill to add an entire browser emulation library just to fix unbalanced HTML but I'm not sure what other solutions are available because direct HTML rendering has been deprecated in mobiledoc and we're unlikely to find an alternative solution that will be as thorough at cleaning up invalid HTML.

File: core/server/apps/default-cards/cards/markdown.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
 var SimpleDom   = require('simple-dom'),
     tokenizer   = require('simple-html-tokenizer').tokenize,
+    jsdom       = require('jsdom').jsdom,
```
Comment (kirrg001) [inline core/server/apps/default-cards/cards/markdown.js]: Could you please move all third party dependencies into the render function?
The background is that we don't have to load them on bootstrap.
This slows down the bootstrap time by ~200ms. Thanks 👍 

Furthermore, the global variables e.g. html, doc can be local variables.

Comment (kevinansfield) [inline core/server/apps/default-cards/cards/markdown.js]: > Could you please move all third party dependencies into the render function?

👍 

> Furthermore, the global variables e.g. html, doc can be local variables.

I tried that but JSCS was complaining so thought it was some node rule I wasn't aware of 😄. I'll make them local and add an ignore if I can't get JSCS to behave

**Comment by kirrg001** [comment]
I did a little memory test, couldn't find any big problem (with > 65k words).
Memory grows to ~500/600mb after saving a lot, but falls down to 100mb after a while.

**Review by kirrg001 — APPROVED** [review]
🙃

**Pull request merged**